doc/charm-structure.rst

```diff
@@ -0,0 +1,67 @@
+Charm Directory and File Structure
```

Comment:
Can we please drop this file for now?  It will be interesting to have something along those lines once we have a firmer design, but for now this will quickly be out of date.

Reply:
OK, just wanted to have a textual description of what I am trying to achieve documented but it does not have to be here initially.